diff --git a/api/api.go b/api/api.go
index bfb94af..6fe6b4c 100644
--- a/api/api.go
+++ b/api/api.go
@@ -288,6 +288,27 @@ func procCreateNetwork(c libnetwork.NetworkController, vars map[string]string, b
 	if len(create.DriverOpts) > 0 {
 		options = append(options, libnetwork.NetworkOptionDriverOpts(create.DriverOpts))
 	}
+
+	ipV4Configs := []*libnetwork.IpamConf{}
+	ipV6Configs := []*libnetwork.IpamConf{}
+	for _, cfg := range create.Ipam.Config {
+		ipConf := &libnetwork.IpamConf{
+			PreferredPool: cfg.Subnet,
+			SubPool:       cfg.IPRange,
+			Gateway:       cfg.Gateway,
+			AuxAddresses:  cfg.AuxAddress,
+		}
+		if ipConf.PreferredPool != "" {
+			if nw, err := types.ParseCIDR(ipConf.PreferredPool); err == nil && nw.IP.To4() == nil {
+				ipV6Configs = append(ipV6Configs, ipConf)
+				continue
+			}
+		}
+		ipV4Configs = append(ipV4Configs, ipConf)
+	}
+
+	options = append(options, libnetwork.NetworkOptionIpam(create.Ipam.Driver, "", ipV4Configs, ipV6Configs))
+
 	nw, err := c.NewNetwork(create.NetworkType, create.Name, options...)
 	if err != nil {
 		return "", convertNetworkError(err)
diff --git a/api/types.go b/api/types.go
index df565c4..cea6f2c 100644
--- a/api/types.go
+++ b/api/types.go
@@ -38,6 +38,7 @@ type networkCreate struct {
 	NetworkType string            `json:"network_type"`
 	DriverOpts  map[string]string `json:"driver_opts"`
 	NetworkOpts map[string]string `json:"network_opts"`
+	Ipam        IPAM              `json:"ipam"`
 }
 
 // endpointCreate represents the body of the "create endpoint" http request message
@@ -78,3 +79,17 @@ type extraHost struct {
 	Name    string `json:"name"`
 	Address string `json:"address"`
 }
+
+// IPAM represents IP Address Management
+type IPAM struct {
+	Driver string       `json:"driver"`
+	Config []IPAMConfig `json:"config"`
+}
+
+// IPAMConfig represents IPAM configurations
+type IPAMConfig struct {
+	Subnet     string            `json:"subnet,omitempty"`
+	IPRange    string            `json:"ip_range,omitempty"`
+	Gateway    string            `json:"gateway,omitempty"`
+	AuxAddress map[string]string `json:"auxiliary_address,omitempty"`
+}
